internal/controller/vintagewc_controller.go

```diff
@@ -81,5 +87,18 @@ func (r *VintageWCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 func (r *VintageWCReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&capiv1beta1.Cluster{}).
+		// This ensures we run the reconcile loop when the observability-bundle app resource version changes.
+		Watches(
+			&appv1alpha1.App{},
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
+				return []reconcile.Request{
+					{NamespacedName: types.NamespacedName{
+						Name:      object.GetLabels()["giantswarm.io/cluster"],
```

Comment:
This creates a reconcile requests for a vintage WC (needs to be tested once we can have a 1.0.0 release)